src/main/scala/com/gu/support/workers/model/RequestInfo.scala

```diff
@@ -1,5 +1,10 @@
 package com.gu.support.workers.model
 
-case class RequestInfo(encrypted: Boolean, testUser: Boolean, failed: Boolean, messages: List[String]){
-  def appendMessage(message: String) = copy(messages = messages :+ message)
+import scala.collection.mutable.ListBuffer
+
+case class RequestInfo(encrypted: Boolean, testUser: Boolean, failed: Boolean, messages: ListBuffer[String] = ListBuffer.empty) {
```

Comment:
Is ListBuffer threadsafe?

We rarely use case classes with mutable state, so I'm worried that at some future point someone will make a faulty assumption based on their expectation of immutability.  Especially in the context of a shared model that is primarily used for Json serialisation mutability is unexpected.

If it is a significant improvement to be able to append to this without making copies, I think it might be better to pass around a (non-case class) `class RequestInfoBuilder` that includes the `appendMessage` method and `def build(): RequestInfo` method

Reply:
@davidfurey I've realised I can use a Writer monad to achieve what I wanted with the existing implementation of RequestInfo so I'm going to close this.